src/main/java/dev/harrel/jsonschema/Applicators.java

```diff
@@ -365,17 +410,28 @@ class OneOfEvaluator implements Applicator {
     }
 
     @Override
-    public boolean apply(EvaluationContext ctx, JsonNode node) {
-        return refs.stream()
-                .filter(uri -> ctx.resolveInternalRefAndValidate(uri, node))
-                .count() == 1;
+    public Set<String> getVocabularies() {
+        return APPLICATOR_VOCABULARY;
+    }
+
+    @Override
+    public Result evaluate(EvaluationContext ctx, JsonNode node) {
+        List<Integer> matchedIndexes = IntStream.range(0, refs.size())
+                .filter(i -> ctx.resolveInternalRefAndValidate(refs.get(i), node))
+                .boxed()
+                .collect(Collectors.toList());
+
+        if (matchedIndexes.size() == 1) {
```

Comment:
Let's also handle case when there are no matched schemas, Suggesting `Value does not match against any of the schemas`

Reply:
Thanks, updated it to suggested error message